nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/FetchS3Object.java

```diff
@@ -46,7 +46,7 @@
 import com.amazonaws.services.s3.model.S3Object;
 
 @SupportsBatching
```

Comment:
Should also be a @TriggerSerially given the state you have below and the inconsistencies that could be generated in multithreaded scenarios.